### ISSUE TITLE
Support Harfbuzz Wasm Shaper

### ISSUE THREAD

**Issue by bbb651** [simoncozens/crowbar#72]
I recently watched this talk [Programmable fonts with Rust by Valdemar Erk](https://youtu.be/Ms1Drb9Vw9M) which showcases cool uses for a relatively new feature in Harfbuzz called Wasm Shaper, which let's font use embedded wasm to programmatically control shaping.

I haven't found a quick way to play around with it, as it seems like it's turned off in all software I use, I think it's off by default and the `-Dwasm=enabled` option enables it. I think it might be a bit tricky to get it to work, in order for Harfbuzz to run wasm when compiled to wasm it either needs to have a wasm runtime in wasm (this is something most runtimes support, but `wasm-micro-runtime` specifically doesn't seem to support it) or to somehow call out to js in order to run it with web apis.

Harfbuzz has [a page](https://github.com/harfbuzz/harfbuzz/blob/main/docs/wasm-shaper.md) with more details and some [demos](https://github.com/harfbuzz/harfbuzz-wasm-examples).

**Comment by simoncozens** [comment]
I wrote the demos. :-)

I haven't been able to compile a WASM version of the WASM runtime, and so I haven't worked out how best to implement this.

**Comment by simoncozens** [comment]
I don't think this is possible, sadly. We would need a WASM-compiled harfbuzz containing its own internal WASM machine.
